### ISSUE TITLE
Research if sov state db key to key-hash mapping is performant

### ISSUE THREAD

**Issue by ercecan** [chainwayxyz/citrea#712]
There is a key to key hash mapping in the statedb, research if it actually makes sense and remove if we are actually better without it

**Comment by eyusufatik** [comment]
@rakanalh do you remember why we cloed the PR closing this?

**Comment by rakanalh** [comment]
@eyusufatik Yes, it needed more time to figure out how to replace trying to fetch a key by name to by hash. We cannot replace key hash of course because JMT uses it.
